cpp/src/reductions/scan.cu

```diff
@@ -217,36 +218,37 @@ template <typename T,
 };
 
 std::unique_ptr<column> scan(const column_view& input,
-                             scan_op op, bool inclusive, bool skipna,
+                             std::unique_ptr<aggregation> const &agg,
+                             bool inclusive, bool skipna,
```

Comment:
`bool`'s in function parameter lists make the call sites very unreadable. I realize this was pre-existing but is it possible to change? Leading to:
 ```cpp
enum class scan_type {
    INCLUSIVE, 
    EXCLSIVE
};

this->scan_test({v.begin(), v.end()}, 
                    {exact.begin(), exact.end()},
                    scan_op::SUM, scan_type::INCLUSIVE);
                    cudf::experimental::make_sum_aggregation(), true);
```
instead of:
```cpp
this->scan_test({v.begin(), v.end()}, 
                    {exact.begin(), exact.end()},
                    scan_op::SUM, true);
                    cudf::experimental::make_sum_aggregation(), true);
```

Reply:
And maybe `skipna` too? Not sure what this is even? Like `skip_na`?

Reply:
Maybe these enums can be centralized along with aggregations. inclusive / exclusive may be useful for rolling operations too (pandas `cumsum`)?  And no doubt other algorithms need to skip nulls?

Reply:
Yes. `skip_nulls` will help. not sure where to put it.
note: legacy groupby_quantiles has `bool include_nulls`. New one does not.

Reply:
> And maybe `skipna` too? Not sure what this is even? Like `skip_na`?

pandas uses `skipna` for excluding nulls in series and dataframe aggregations. 
it is not used by groupby and rolling. I tried pandas grouby and rolling - mean, sum, count ec. They don't accept skipna as kwarg.